src/foam/dao/FileJournal.js

```diff
@@ -281,27 +285,58 @@ foam.CLASS({
           .toString());
       `
     },
+    {
+      name: 'getEntry',
+      documentation: 'Reads a syntatically meaningful unit from the journal',
+      type: 'String',
+      args: [
+        {
+          name: 'reader',
+          type: 'BufferedReader'
+        }
+      ],
+      javaCode: `
+        try {
+          String line = reader.readLine();
+          if ( getMultiLine() ) {
+            StringBuilder sb = new StringBuilder();
+            while ( ! line.endsWith("})") ) {
```

Comment:
I think we actually want to check line.equals("})"), this will help protect against lines in js scripts like "var person = Person.create({ name: "adam" })"

Reply:
A proper multiline journal entry shoudl read

```p({"class":"foam.nanos.script.Script","code":"""doo
some

stuff""", "otherProperty":123
})
```